### PR TITLE
(x/mint) Use new BeginBlock extension

### PR DESCRIPTION
**Pull request by giansalex**

### CODE REVIEW

**Comment by the-frey** [comment]
Sorry, need this one ELI5'd if that's okay.

**Comment by the-frey** [comment]
Still need the change explained - probably me being dumb but what's the actual impact?

**Comment by giansalex** [comment]
is just a refactoring, so that the modules don't have to add BeginBlock/EndBlock functions they will not use.

**Review by the-frey — APPROVED** [review]

**Review by faddat — APPROVED** [review]
